src/esp/sim/BatchReplayRenderer.cpp

```diff
@@ -106,6 +108,49 @@ BatchReplayRenderer::BatchReplayRenderer(
           Mn::Matrix4::from(rotation.toMatrix(), translation);
     }
 
+    void changeLightSetup(const esp::gfx::LightSetup& lights) override {
+      if (!renderer_.maxLightCount()) {
+        ESP_WARNING() << "Attempted to change" << lights.size()
+                      << "lights for scene" << sceneId_
+                      << "but the renderer is configured without lights";
+        return;
+      }
+
+      renderer_.clearLights(sceneId_);
+      for (std::size_t i = 0; i != lights.size(); ++i) {
+        const gfx::LightInfo& light = lights[i];
+        CORRADE_INTERNAL_ASSERT(light.model == gfx::LightPositionModel::Global);
+
+        const std::size_t nodeId = renderer_.addEmptyNode(sceneId_);
+
+        /* Clang Tidy, you're stupid, why do you say that "lightId" is not
```

Comment:
>an accidentally zero-initialized variable is *also* a bug, you know?

wut

Reply:
@mosra Just add NOLINTNEXTLINE comment then. :)

Reply:
Because if the variable was meant to be initialized to something, but by accident wasn't, the zero-initialization would hide that bug, causing the program to behave seemingly fine but incorrect. While not having it initialized at all (kept as a random memory) could possibly lead to a random OOB assertion down the line, or cause the program to behave randomly, making it easier to discover the bug.

At least in my experience it was always like this, bugs caused by random memory were easier to discover than bugs where the variables were zero-initialized by accident.

Reply:
>cause the program to behave randomly, making it easier to discover the bug

Not my experience. Nondeterministic bugs are scary. E.g. by leaving it uninitialized, it gets initialized to zero in debug builds but something random and nonzero in release builds, so the entire EA Sports Nascar 2005 eng team spends the entire afternoon tracking down why only the release build is broken. :)

Reply:
Counterpoint to a counterpoint: GCC 12 has [`-ftrivial-auto-var-init=pattern`](https://gcc.gnu.org/onlinedocs/gcc-12.2.0/gcc/Optimize-Options.html#index-ftrivial-auto-var-init) which initializes local variables in debug builds to some bit pattern. MemorySanitizer can do something similar for the heap as well, and maybe for local variables too.

So if the variable is accidentally left uninitialized, these options can be used to make the bug behave deterministically. But if it gets zero-initialized because Clang Tidy demanded to, and then accidentally not set to the value it should have, it's harder to discover, as it would behave as a "regular" value and won't trigger any assertions or sanitizer reports. In my opinion. But my general opinion is fail fast & fail hard, which is probably something completely undesirable in game development :)

Reply:
If this were my code, I'd still prefer to initialize to an aggressively-invalid value like -1 or something, but I don't feel too strongly and you've clearly given this some thought. Fine to leave this as-is!